### DISCUSSION TITLE
Bug? Generated code can't reference types from SDK (ASP.NET Core)

### DISCUSSION THREAD

**Discussion by mkane91301** [dotnet/roslyn#62556] [Q&A]
I've made a Roslyn code generator that generates the implementation of a partial method that takes and returns a Microsoft.AspNetCore.Routing.IEndpointRouteBuilder (and the code uses the fully-qualified type name rather than a using directive). Yet, when I have it generate code in an ASP.NET Core project, the compiler complains that the partial declaration has no implementation and the partial implementation has no declaration. Examining the generated code in Visual Studio, it's clear from the syntax highlighting that the compiler doesn't see the Microsoft.AspNetCore namespace in the context of the generated code, so it doesn't know what this type is.

Is this a bug, a known limitation of code generation, or is there something else I need to do to make this work?

**Comment by jasonmalinowski** [comment]
> Examining the generated code in Visual Studio, it's clear from the syntax highlighting that the compiler doesn't see the Microsoft.AspNetCore namespace in the context of the generated code, so it doesn't know what this type is.

Can you share a screenshot? There might be a few hints we can figure out from that.

**Reply by mkane91301** [reply]
Here's the csproj file for the project using the source generator:
![image](https://user-images.githubusercontent.com/2616652/178543955-a506a2b7-3b0d-455b-bcfb-394cec687bb2.png)
Here's the hand-written code triggering the generation:
![image](https://user-images.githubusercontent.com/2616652/178544147-4bfbb0bb-3ea8-4fc0-be2e-af8df44593c1.png)
Here's the generated code:
![image](https://user-images.githubusercontent.com/2616652/178544254-52389bd2-32da-4c6e-8c75-42afb81e69eb.png)

**Reply by Youssef1313** [reply]
@mkane91301 It looks like your generated code is producing wrong namespace. So your two `RoutingExtensions` classes are really very different.

**Reply by mkane91301** [reply]
Oh, dammit, yeah, I just needed more eyeballs on it. I'm only outputting the last part of the namespace instead of the fully-qualified namespace! Thanks!

**Comment by Youssef1313** [comment]
It would be good to know if the errors show in the IDE only? or also from a command-line build?

**Reply by mkane91301** [reply]
The build errors are identical whether I build in VS or dotnet build. I get CS8795 on the partial method declaration and CS0759 on the partial method implementation in the generated code. But I DO NOT get any error about the compiler not recognizing Microsoft.AspNetCore.Routing.IEndpointRouteBuilder in the generated code. It's only in the IDE that it shows without the syntax highlighting that it would have if it recognized the namespace and type.
